src/data/users.js

```diff
@@ -0,0 +1,146 @@
+const mongoCollections = require('../config/mongoCollections');
+const boards = mongoCollections.boards;
+const users = mongoCollections.users;
+const {user_colors} = require('../public/constants/index');
+const {ObjectId} = require('mongodb');
+const bcrypt = require('bcrypt');
+const saltRounds = 16;
+const error_handler = require('../errors/error-handler'); 
+
+
+module.exports = {
+  /**
+   * Get a user with a specific ID.
+   * @param {String} id The user's id.
+   * @returns A user.
+   */
+  readByID: async (id) => {
+    if(!error_handler.checkObjectId(id))
+      throw new Error("id is not valid.");
+
+     const userCollection = await users();
+     const user = await userCollection.findOne({_id: ObjectId(id)});
+     if(user === null)
+       throw new Error("There is no user with that id.");
+ 
+     user['_id'] = user['_id'].toString();
+ 
+     return user;
+  },
+
+  /** Don't need a readAll for users? */
+
+  /**
+   * Adds a new user. Picks a random color from pre-approved values, hashes the password, and starts the list of boards as empty.
+   * @param {String} email The email address of the user.
+   * @param {String} firstName The first name of the user.
+   * @param {String} lastName The last name of the user.
+   * @param {String} password The password chosen by the user that will be hashed.
+   * @returns The user object.
+   */
+  create: async (email, firstName, lastName, password) => {
+    if(!error_handler.checkEmail(email))
+      throw new Error("Email is not valid.");
+      
+    if(!error_handler.checkFirstName(firstName))
+      throw new Error("First name is not valid.");
+
+    if(!error_handler.checkLastName(lastName))
+      throw new Error("Last name is not valid.");
+
+    if(!error_handler.checkNonEmptyString(password))
+      throw new Error("Password is not valid.");
+
+    const hashedPassword = await bcrypt.hash(password, saltRounds);
+    const color = user_colors[Math.floor(Math.random() * user_colors.length)];
+
+    let newUser = {
+      email,
+      firstName,
+      lastName,
+      color,
+      hashedPassword
+    };
+    newUser.boards = [];
```

Comment:
not sure why this is a separate statement outside. not necessary to change, just unsure on why it is like this

Reply:
It's separate because "boards" is already a variable